game/src/ecs/components/ai/AITools.java

```diff
@@ -0,0 +1,121 @@
+                    level.findPath(
+                            level.getTileAt(position.toCoordinate()), level.getTileAt(newPosition));
+
+            System.out.println("PATH " + path.getCount());
+            return path;
+        }
+        return null;
+    }
+
+    public static GraphPath<Tile> calculateNewPath(Entity entity, Entity entity2) {
+        PositionComponent myPositionComponent =
+                (PositionComponent) entity.getComponent(PositionComponent.name);
+        PositionComponent heroPositionComponent =
+                (PositionComponent) entity2.getComponent(PositionComponent.name);
+        if (myPositionComponent != null && heroPositionComponent != null) {
+            ILevel level = ECS.currentLevel;
+            Coordinate myPosition = myPositionComponent.getPosition().toCoordinate();
+            Coordinate heroposition = heroPositionComponent.getPosition().toCoordinate();
+            return level.findPath(level.getTileAt(myPosition), level.getTileAt(heroposition));
+        }
+        return null;
+    }
+
+    public static void move(Entity entity, GraphPath<Tile> path) {
+        PositionComponent pc = (PositionComponent) entity.getComponent(PositionComponent.name);
+        VelocityComponent vc = (VelocityComponent) entity.getComponent(VelocityComponent.name);
+        ILevel level = ECS.currentLevel;
+        Tile currentTile = level.getTileAt(pc.getPosition().toCoordinate());
+        int i = 0;
+        Tile nextTile = null;
+        do {
+            if (i >= path.getCount()) return;
+            if (path.get(i).equals(currentTile)) {
+                nextTile = path.get(i + 1);
+            }
+            i++;
+        } while (nextTile == null);
+        switch (currentTile.directionTo(nextTile)[0]) {
+            case N:
+                vc.setY(vc.getySpeed());
+                break;
+            case S:
+                vc.setY(-vc.getySpeed());
+                break;
+            case E:
+                vc.setX(vc.getxSpeed());
+                break;
+            case W:
+                vc.setX(-vc.getxSpeed());
+                break;
+        }
+        if (currentTile.directionTo(nextTile).length > 1)
+            switch (currentTile.directionTo(nextTile)[1]) {
+                case N:
+                    vc.setY(vc.getySpeed());
+                    break;
+                case S:
+                    vc.setY(-vc.getySpeed());
+                    break;
+                case E:
+                    vc.setX(vc.getxSpeed());
+                    break;
+                case W:
+                    vc.setX(-vc.getxSpeed());
+                    break;
+            }
+    }
+
+    public static boolean inRange(Entity entity, float range) {
+        PositionComponent myPositionComponent =
+                (PositionComponent) entity.getComponent(PositionComponent.name);
+        if (ECS.hero != null) {
+            PositionComponent heroPositionComponent = ECS.hero.getPositionComponent();
```

Comment:
Hier wird nach `hero.getPositionComponent()` gefragt. Oben in `calculateNewPath` wird `entity.getComponent(BlaComponent.name)` genutzt. Gibt es einen Grund, warum das nicht uniform gemacht wird?

Reply:
siehe #130